internal/cuckoo/cuckoo_test.go

```diff
@@ -39,31 +39,37 @@ func TestNewCuckoo(t *testing.T) {
 	seeds := makeSeeds()
 
 	for _, tt := range cuckooTests {
-		c := NewCuckoo(tt.size, seeds)
-		if c.bucketSize != tt.bSize {
-			t.Errorf("cuckoo bucketsize: want: %d, got: %d", tt.bSize, c.bucketSize)
+		c, err := NewCuckoo(tt.size, seeds)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if c.CuckooHasher.bucketSize != tt.bSize {
+			t.Errorf("cuckoo bucketsize: want: %d, got: %d", tt.bSize, c.CuckooHasher.bucketSize)
 		}
 	}
 }
 
 func TestInsertAndGetHashIdx(t *testing.T) {
-	cuckoo := NewCuckoo(testN, makeSeeds())
+	cuckoo, err := NewCuckoo(testN, makeSeeds())
+	if err != nil {
+		t.Fatal(err)
+	}
 	errCount := 0
 	testData := genBytes(int(testN))
 
 	insertTime := time.Now()
-	for idx, item := range testData {
-		if err := cuckoo.insert(uint64(idx+1), item); err != nil {
+	for _, item := range testData {
+		if err := cuckoo.Insert(item); err != nil {
 			errCount += 1
 		}
 	}
 
 	t.Logf("To be inserted: %d, bucketSize: %d, load factor: %f, failure insertion:  %d, taken %v",
-		testN, cuckoo.bucketSize, cuckoo.LoadFactor(), errCount, time.Since(insertTime))
+		testN, cuckoo.CuckooHasher.bucketSize, cuckoo.LoadFactor(), errCount, time.Since(insertTime))
```

Comment:
can you do `cuckoo.bucketSize` directly?

Reply:
yes